File: py_ws/src/py_pkg/py_pkg/subscriber_node.py
import rclpy

from rclpy.node import Node
from std_msgs.msg import String
from std_msgs.msg import Float32

topic_1 = 'float_topic'
topic_2 = 'string_topic'

class SubscriberNode(Node):
    def __init__(self):
        super().__init__('subscriber_node')
        self.subscription_float = self.create_subscription(Float32, topic_1, self.float_callback, 10)
        self.subscription_string = self.create_subscription(String, topic_2, self.string_callback, 10)

    def float_callback(self, msg):
        self.get_logger().info(f'Received float: {msg.data}')

    def string_callback(self, msg):
        self.get_logger().info(f'Received string: {msg.data}')
def main(args=None):
    rclpy.init(args=args)
    node = SubscriberNode()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()
if __name__ == '__main__':
    main()

File: cpp_ws/src/cpp_pkg/src/subscriber_node.cpp
#include "rclcpp/rclcpp.hpp"

#include "std_msgs/msg/string.hpp"
#include "std_msgs/msg/float32.hpp"


#include <memory>


// This is a simple ROS 2 publisher node that publishes messages to a topic named "topic".
// It publishes a string message every 500 milliseconds, incrementing a count each time.
using namespace std::chrono_literals;

auto topic_1 = "float_topic";
auto topic_2 = "string_topic";

using std::placeholders::_1; 

class SubscriberNode : public rclcpp::Node
{
public:
    SubscriberNode() : Node("subscriber_node")
    {
        subscriber_float_ = this->create_subscription<std_msgs::msg::Float32>(
            topic_1, 10, std::bind(&SubscriberNode::float_callback, this, _1));
        subscriber_string_ = this->create_subscription<std_msgs::msg::String>(
            topic_2, 10, std::bind(&SubscriberNode::string_callback, this, _1));

        RCLCPP_INFO(this->get_logger(), "SubscriberNode has been started.");
    }
    void float_callback(const std_msgs::msg::Float32::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received float message: '%f'", msg->data);
    }
    void string_callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received string message: '%s'", msg->data.c_str());
    }
private:
    rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr subscriber_float_;
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_string_;
};
int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SubscriberNode>());
    rclcpp::shutdown();
    return 0;
}
// This code defines a ROS 2 subscriber node that listens to two topics: "float_topic" and "string_topic".